python/ray/_private/runtime_env/path_utils.py

```diff
@@ -0,0 +1,71 @@
+import os
+
+from typing import List, Optional
+from ray._private.runtime_env import env_utils
+
+
+INTERNAL_PIP_FILENAME = "ray_runtime_env_internal_pip_requirements.txt"
+MAX_INTERNAL_PIP_FILENAME_TRIES = 100
+
+
+class PathHelper:
+    @staticmethod
+    def get_virtualenv_path(target_dir: str) -> str:
+        return os.path.join(target_dir, "virtualenv")
+
+    @classmethod
+    def get_virtualenv_python(cls, target_dir: str) -> str:
+        virtualenv_path = cls.get_virtualenv_path(target_dir)
+        if env_utils._WIN32:
+            return os.path.join(virtualenv_path, "Scripts", "python.exe")
+        else:
+            return os.path.join(virtualenv_path, "bin", "python")
+
+    @classmethod
+    def get_virtualenv_activate_command(cls, target_dir: str) -> List[str]:
+        virtualenv_path = cls.get_virtualenv_path(target_dir)
+        if env_utils._WIN32:
+            cmd = [os.path.join(virtualenv_path, "Scripts", "activate.bat")]
+
+        else:
+            cmd = ["source", os.path.join(virtualenv_path, "bin/activate")]
+        return cmd + ["1>&2", "&&"]
+
+    @staticmethod
+    def get_requirements_file(target_dir: str, pip_list: Optional[List[str]]) -> str:
```

Comment:
should this be under `dependency_utils.py`

Reply:
Kinda tricky, for this particular function, it's still returning path to dependency.

Reply:
I put it under `path_utils` just to keep the old interface: https://github.com/ray-project/ray/blob/b2fafa38277dd70611342489f4cf72bf0628cade/python/ray/_private/runtime_env/pip.py#L74-L111
(maybe we could address later?)

Reply:
Ok. I was thinking to get rid of `path_utils.py` and put things into the other two utils files but its not a big deal.

Reply:
Completely removed path utils.